apps/versions/models.py

```diff
@@ -48,10 +48,23 @@ def get_query_set(self):
 
 
 def source_upload_path(instance, filename):
+    VALID_SOURCE_EXTENSIONS = (
+        '.zip', '.tar', '.7z', '.tar.gz', '.tgz', '.tbz', '.txz', '.tar.bz2',
+        '.tar.xz'
+    )
+    file_ext = ''
+    for ext in VALID_SOURCE_EXTENSIONS:
+        if filename.endswith(ext):
+            file_ext = ext
```

Comment:
You actually don't need to do this. When you break (or end) a loop, `ext` will be assigned to its latest value. So if you remove `file_ext = ''` and remove the affectation on line 58, you can simply use `ext` on line 67.